File: static/vector.js
class Vector{
	
	constructor(x,y){
		this._x = x;
		this._y = y;
		var obj = Object.create(this);
		obj.setX(x);
		obj.setY(y);
		return obj;
	}

	setX(value){
		this._x = value;
	}

	setY(value){
		this._y = value;
	}

	getX(){
		return this._x;
	}

	getY(){
		return this._y;
	}


	setAngle(angle){
		var length = this.getLength();
		this._x = Math.cos(angle) * length;
		this._y = Math.sin(angle) * length;
	}


	getAngle(){
		return Math.atan2(this._y, this._x);
	}

	getLength(){
		return Math.sqrt(this._x * this._x + this._y * this._y);
	}


	setLength(length){
		var angle = this.getAngle();
		this._x = Math.cos(angle) * length;
		this._y = Math.sin(angle) * length;
	}


	add(v2){
		return vector.create(this._x+v2.getX(), this._y+v2.getY());
	}

	subtract(v2){
		return vector.create(this._x-v2.getX(), this._y-v2.getY());
	}

	multiply(val){
		return vector.create(this._x*val, this._y*val);
	}

	divide(val){
		return vector.create(this._x/val, this._y/val);
	}


	addTo(v2){
		this._x += v2.getX();
		this._y += v2.getY();
	}

	
	subtractFrom(v2){
		this._x -= v2.getX();
		this._y -= v2.getY();
	}	

	multiplyBy(val){
		this._x *=val;
		this._y *=val;
	}

	divideBy(val){
		this._x /= val;
		this._y /= val;
	}


};

File: vector.py
import math

class Vector:

	def __init__(self, x, y):
		self.x = x
		self.y = y

	def setX(self,value):
		self.x = value

	def getX(self):
		return self.x

	def setY(self,value):
		self.y = value

	def getY(self):
		return self.y

	def setAngle(self,angle):
		length = self.getLength()
		self.x = math.cos(angle) * length
		self.y = math.sin(angle) * length


	def getAngle(self):
		return math.atan2(self.y,self.x)

	def getLength(self):
		return math.sqrt(self.x*self.x + self.y*self.y)


	def setLength(self,length):
		angle = self.getAngle()
		self.x = math.cos(angle) * length
		self.y = math.sin(angle) * length


	def add(self, v2):
		return Vector(self.x+v2.getX(), self.y+v2.getY())


	def subtract(self, v2):
		return Vector(self.x-v2.getX(), self.y-v2.getY())


	def multiply(self,val):
		return Vector(self.x*val, self.y*val)

	def divide(self,val):
		return Vector(self.x/val, self.y/val)

	def addTo(self,v2):
		self.x += v2.getX()
		self.y += v2.getY()


	def subtractFrom(self,v2):
		self.x -= v2.getX()
		self.y -= v2.getY()


	def multiplyBy(self, val):
		self.x *= val
		self.y *= val

	def divideBy(self,val):
		self.x /= val
		self.y /= val



